src/components/interview.ts

```diff
@@ -35,177 +42,86 @@ function parseLink(link: string) {
-    return;
+  } else if (!(domain in availableDomains)) {
+    // domain not a valid key in availableDomains
+    throw 'Invalid domain.';
   } else {
+    // query interviewers by domain
     res = await db.all(
       'SELECT * FROM interviewers WHERE user_id IN (SELECT user_id FROM domains WHERE domain = ?)',
       domain
     );
-    title = 'Available interviewers for ' + available_domains[domain];
   }
 
-  //shuffles and prints the results thru embed
-  res = _.shuffle(res) as Interviewer[];
-  const outEmbed = new Discord.MessageEmbed().setColor('#0099ff').setTitle(title);
-  let listString = '';
-  for (let count = 0; count < RESULTS_PER_PAGE && count < res.length; count++) {
-    const rows = res[count];
-    listString +=
-      '**' + (await client.users.fetch(rows['user_id'].toString())).tag + '** | [Calendar](' + rows['link'] + ')\n\n';
-  }
-  outEmbed.setDescription(listString);
-  await message.reply(outEmbed);
-}
+  return res;
+};
 
-async function clearProfile(message: Discord.Message): Promise<void> {
+export const clearProfile = async (id: string): Promise<void> => {
   const db = await openDB();
-  const { id } = message.author;
 
   //clear user data from both tables
-  await db.run('DELETE FROM interviewers WHERE user_id = ?', id);
   await db.run('DELETE FROM domains WHERE user_id = ?', id);
-  await message.reply('Your interviewer data has been cleared!');
-}
+  await db.run('DELETE FROM interviewers WHERE user_id = ?', id);
+};
 
-async function showProfile(message: Discord.Message): Promise<void> {
+export const getDomains = async (id: string): Promise<string[]> => {
   const db = await openDB();
-  const { id } = message.author;
-
-  //check if user signed up to be interviewer
-  const interviewer = await getInterviewer(id);
-  if (!interviewer) {
-    await message.reply("You don't seem to have signed up yet, run `.interviewer signup [link]`");
-    return;
-  }
 
   //get user's domains (if any)
-  const res = await db.all('SELECT * FROM domains WHERE user_id = ?', id);
-  const domains = _.join(
-    res.map((x) => available_domains[x.domain]),
-    ', '
-  );
-
-  //build output embed
-  const outEmbed = new Discord.MessageEmbed().setColor('#0099ff').setTitle('Interviewer Profile');
-  outEmbed.addField('**Link**', interviewer['link']);
-  outEmbed.addField('**Domains**', domains === '' ? 'None' : domains);
-  await message.reply(outEmbed);
-}
+  const res = await db.all('SELECT domain FROM domains WHERE user_id = ?', id);
+  return res.map((row) => availableDomains[row.domain]);
+};
 
-async function addDomain(message: Discord.Message, args: string[]): Promise<void> {
+/*
+  If domain already exists, remove interivewer from the domain and return true.
+  If domain doesn't exist, add interviewer to the domain and return false.  
+  Throws an error if domain isn't provided or not valid.
+*/
+export const toggleDomain = async (id: string, domain: string): Promise<boolean> => {
```

Comment:
I think we only care if inDomain carries a value, but should it be a boolean? Or is it like that on purpose to kinda map empty to 0 and non-empty to 1...

Reply:
yeah this just casts to bool, we could do a `!!inDomain` if it makes it clearer

Reply:
oh it actually doesn't, I'll have to cast it explicitly, good catch :)